src/forms/medication-dispense-review.component.tsx

```diff
@@ -164,36 +135,27 @@ const MedicationDispenseReview: React.FC<MedicationDispenseReviewProps> = ({
   );
 
   // check to see if the current dispense would be a substitution, and update accordingly
-  useEffect(() => {
-    if (
-      medicationRequest?.medicationReference?.reference &&
-      medicationDispense?.medicationReference?.reference &&
-      medicationRequest.medicationReference.reference != medicationDispense.medicationReference.reference
-    ) {
-      setIsSubstitution(true);
-      updateMedicationDispense({
-        ...medicationDispense,
-        substitution: {
-          ...medicationDispense.substitution,
-          wasSubstituted: true,
-        },
-      });
+  const isSubstitution =
+    medicationRequest?.medicationReference?.reference &&
+    medicationDispense?.medicationReference?.reference &&
+    medicationRequest.medicationReference.reference != medicationDispense.medicationReference.reference;
+  const substitution = useMemo(() => {
+    if (isSubstitution) {
+      // make sure that the value medicationDispense.substitution.wasSubstituted exists and is truthy
+      if (medicationDispense.substitution.wasSubstituted) {
+        return medicationDispense.substitution;
+      } else {
+        return { ...medicationDispense.substitution, wasSubstituted: true };
+      }
     } else {
-      setIsSubstitution(false);
-      updateMedicationDispense({
-        ...medicationDispense,
-        substitution: {
-          wasSubstituted: false,
-          reason: [
-            {
-              coding: [{ code: null }],
-            },
-          ],
-          type: { coding: [{ code: null }] },
-        },
-      });
+      return blankSubstitution;
     }
-  }, [medicationDispense, medicationRequest, updateMedicationDispense]);
+  }, [isSubstitution, medicationDispense]);
```

Comment:
```suggestion
  }, [isSubstitution, medicationDispense.substitution]);
  
```

Can we change the substitution deps  in the deps array from `medicationDispense` to `medicationDispense.substitution`? This should prevent recalculation when quantity/dosage changes.